### ISSUE TITLE
GPU processing thread is too slow, waiting on CPU.

### ISSUE THREAD

**Issue by dexter74** [Ryujinx/Ryujinx-Games-List#4624]
### Game Name

Mario Mansion 3 

### Game Update Version

1.4.0 (Luigis Mansion 3 - v1.4.0.xci)

### Emulator Build Version

1.1.810 / 1.1.814

### Graphics Backend

Vulkan

### Hardware Specs

- ##### CPU: *AMD Ryzen 7 3700X*
- ##### GPU: *AMD RX 6700XT*
- ##### RAM: *4 x 8 Go DDR4 (3200MHz)*

### Game State

Crash in game. (With or without mod)

### Log File
```
00:00:00.055 |I| Configuration LogValueChange: ResScale set to: 1
00:00:00.060 |I| Configuration LogValueChange: ResScaleCustom set to: 1
00:00:00.060 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
00:00:00.061 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
00:00:00.062 |I| Configuration LogValueChange: BackendThreading set to: On
00:00:00.063 |I| Configuration LogValueChange: GraphicsBackend set to: Vulkan
00:00:00.063 |I| Configuration LogValueChange: PreferredGpu set to: 0x1002_0x73DF
00:00:00.064 |I| Configuration LogValueChange: AntiAliasing set to: None
00:00:00.064 |I| Configuration LogValueChange: ScalingFilter set to: Bilinear
00:00:00.064 |I| Configuration LogValueChange: ScalingFilterLevel set to: 80
00:00:00.067 |I| Configuration LogValueChange: EnableDockedMode set to: True
00:00:00.067 |I| Configuration LogValueChange: EnableVsync set to: True
00:00:00.067 |I| Configuration LogValueChange: EnableShaderCache set to: True
00:00:00.067 |I| Configuration LogValueChange: EnableTextureRecompression set to: False
00:00:00.067 |I| Configuration LogValueChange: EnableMacroHLE set to: True
00:00:00.067 |I| Configuration LogValueChange: EnablePtc set to: True
00:00:00.068 |I| Configuration LogValueChange: EnableInternetAccess set to: False
00:00:00.068 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
00:00:00.068 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
00:00:00.068 |I| Configuration LogValueChange: AudioBackend set to: SDL2
00:00:00.068 |I| Configuration LogValueChange: AudioVolume set to: 1
00:00:00.069 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
00:00:00.069 |I| Configuration LogValueChange: ExpandRam set to: False
00:00:00.069 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
00:00:00.069 |I| Configuration LogValueChange: UseHypervisor set to: True
00:00:00.078 |N| Application PrintSystemInfo: Ryujinx Version: 1.1.810
00:00:00.083 |N| Application Print: Operating System: Microsoft Windows 10.0.19045 (X64)
00:00:00.083 |N| Application Print: CPU: AMD Ryzen 7 3700X 8-Core Processor ; 8 logical
00:00:00.084 |N| Application Print: RAM: Total 20442 MiB ; Available 15483 MiB
00:00:00.086 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest
00:00:00.087 |N| Application PrintSystemInfo: Launch Mode: UserProfile
00:00:01.322 |W| Application CheckScreenBounds: Failed to find valid start-up coordinates. Defaulting to primary monitor center.
00:00:00.179 |I| Gpu : Backend Threading (On): True
00:00:00.484 |N| Application LoadGuestApplication: Using Firmware Version: 16.0.2
00:00:00.484 |I| Application LoadGuestApplication: Loading as XCI.
00:00:00.500 |I| Application AddAocItem: Found AddOnContent with TitleId 0100DCA0064A7001
00:00:00.504 |I| Application AddAocItem: Found AddOnContent with TitleId 0100DCA0064A7002
00:00:00.526 |I| ModLoader QueryContentsDir: Searching mods for Title 0100DCA0064A6000
00:00:00.528 |I| ModLoader AddModsFromDirectory: Found mod '<0100dca0064a6000 ExeFs>' [E]
00:00:00.529 |I| ModLoader QueryContentsDir: Searching mods for DLC 0100DCA0064A7001
00:00:00.529 |I| ModLoader QueryContentsDir: Searching mods for DLC 0100DCA0064A7002
00:00:00.530 |I| ModLoader QueryContentsDir: Searching mods for Title 0100DCA0064A6000
00:00:00.530 |I| ModLoader QueryContentsDir: Searching mods for DLC 0100DCA0064A7001
00:00:00.530 |I| ModLoader QueryContentsDir: Searching mods for DLC 0100DCA0064A7002
00:00:00.533 |I| Loader Load: Loading rtld...
00:00:00.541 |I| Loader PrintRoSectionInfo: rtld:
    Module: nnrtld
00:00:00.541 |I| Loader Load: Loading main...
00:00:00.669 |I| Loader PrintRoSectionInfo: main:
    Module: D:\Project\NeoPlum\Out\Dist\NXROM\NeoPlumNX64_dist.nss
    SDK Libraries: SDK MW+Microsoft+Havok_Cloth-2015_1
                   SDK MW+Microsoft+Havok-2015_1
                   SDK MW+Microsoft+Havok_Physics2012-2015_1
                   SDK MW+Nintendo+PiaCommon-5_17_1
                   SDK MW+Nintendo+Pia-5_17_1
                   SDK MW+Nintendo+PiaLocal-5_17_1
                   SDK MW+Nintendo+PiaNex-5_17_1-forNEX-4_6_2
                   SDK MW+Nintendo+PiaSession-5_17_1
                   SDK MW+Nintendo+PiaTransport-5_17_1
                   SDK MW+Nintendo+NEX_MM-4_6_2
                   SDK MW+Nintendo+NEX-4_6_2-
                   SDK MW+Nintendo+NEX_CO-4_6_2
                   SDK MW+Nintendo+NEX_SS-4_6_2
                   SDK MW+Nintendo+NintendoSDK_libcurl-7_3_2-Release
                   SDK MW+Nintendo+NintendoSDK_libz-7_3_2-Release
00:00:00.669 |I| Loader Load: Loading subsdk0...
00:00:00.723 |I| Loader PrintRoSectionInfo: subsdk0:
    Module: nnSdkEn
    FS SDK Version: 7.3.2
    SDK Libraries: SDK MW+Nintendo+NintendoSDK_libz-7_3_2-Release
                   SDK MW+Nintendo+NintendoSdk_nnSdk-7_3_2-Release
                   SDK MW+Nintendo+NintendoSDK_NVN-7_3_2-Release
00:00:00.723 |I| Loader Load: Loading subsdk1...
00:00:00.755 |I| Loader PrintRoSectionInfo: subsdk1:
    Module: multimedia
    SDK Libraries: SDK MW+Nintendo+NintendoSDK_movie-7_3_2-Release
00:00:00.755 |I| Loader Load: Loading sdk...
00:00:00.807 |I| Loader PrintRoSectionInfo: sdk:
    Module: nnSdk
    FS SDK Version: 7.3.2
    SDK Libraries: SDK MW+Nintendo+NintendoSDK_libz-7_3_2-Release
                   SDK MW+Nintendo+NintendoSdk_nnSdk-7_3_2-Release
                   SDK MW+Nintendo+NintendoSDK_NVN-7_3_2-Release
00:00:00.817 |I| ModLoader ApplyProgramPatches: Matching IPSwitch patch '1.4.0.pchtxt' in '<0100dca0064a6000 ExeFs>' bid=79E5950FFA85ACF63E28C9AEC051EAC27D6F7F8D
00:00:00.818 |I| ModLoader Parse: IPSwitch:    # Luigi's Mansion 3 [0100DCA0064A6000]- v1.4.0 60FPS
00:00:00.818 |I| ModLoader Parse: IPSwitch:    print_values 0xb7aed0 <= 21008052
00:00:00.819 |I| ModLoader Parse: IPSwitch:    print_values 0x13c95dc <= 00008041
00:00:00.827 |I| ModLoader Patch: Patching address offset b7add0 <= 21 00 80 52 len=4
00:00:00.827 |I| ModLoader Patch: Patching address offset 13c94dc <= 00 00 80 41 len=4
00:00:01.000 |I| Ptc Initialize: Initializing Profiled Persistent Translation Cache (enabled: True).
00:00:01.270 |I| Ptc Load: Loaded Translation Cache (size: 19699550 bytes, translated functions: 13696).
00:00:01.282 |I| Ptc Load: Loaded Profiling Info (size: 43255 bytes, profiled functions: 13467).
00:00:01.292 |I| Loader LoadNsos: Loading image 0 at 0x0000000008000000...
00:00:01.295 |I| Loader LoadNsos: Loading image 1 at 0x0000000008004000...
00:00:01.304 |I| Loader LoadNsos: Loading image 2 at 0x0000000009d1b000...
00:00:01.308 |I| Loader LoadNsos: Loading image 3 at 0x000000000a9ef000...
00:00:01.310 |I| Loader LoadNsos: Loading image 4 at 0x000000000b10c000...
00:00:01.317 |I| ModLoader LoadCheats: Build ids found for title 0100DCA0064A6000:
    0B30BEE8500B0002A171757E16F1E754BA0216A5000000000000000000000000
    79E5950FFA85ACF63E28C9AEC051EAC27D6F7F8D000000000000000000000000
    DBB54B5AB0B60136E02025DDBAB05F698DB9FC00000000000000000000000000
    95501E203C3F39918FCC7C25FEE7CC84FE494A92000000000000000000000000
    FCF93A41B08C5193A1410D262602DC621617C56F000000000000000000000000
00:00:01.320 |I| Application EnsureSaveData: Ensuring required savedata exists.
00:00:01.385 |I| Loader Start: Application Loaded: Luigi's Mansion 3 v1.4.0 [0100dca0064a6000] [64-bit]
00:00:01.439 |I| GUI.WindowThread Hid Configure: Configured Controller ProController to Player1
00:00:01.462 |I| GUI.WindowThread Hid SetupNpad: Connected Controller ProController to Player1
00:00:01.688 |N| GUI.RenderThread Gpu PrintGpuInformation: AMD AMD Radeon RX 6700 XT (Vulkan v1.3.246, Driver v2.0.262)
00:00:01.700 |I| GPU.MainThread Gpu LoadShaders: Loading 98 shaders from the cache...
00:00:02.307 |I| GPU.MainThread Gpu LoadShaders: Rebuilding 98 shaders...
00:00:02.786 |I| GPU.MainThread Gpu LoadShaders: Rebuilt 98 shaders successfully.
00:00:02.787 |I| GPU.MainThread Gpu LoadShaders: Shader cache loaded.
00:00:03.279 |I| HLE.GuestThread.21 Ptc LoadTranslations: 13467 translated functions loaded
00:00:04.448 |I| HLE.OsThread.8 Loader OpenDataStorageByDataId: Opened AddOnContent Data TitleID=0100DCA0064A7001
00:00:04.452 |I| HLE.OsThread.8 Loader OpenDataStorageByDataId: Opened AddOnContent Data TitleID=0100DCA0064A7001
00:00:04.455 |I| HLE.OsThread.8 Loader OpenDataStorageByDataId: Opened AddOnContent Data TitleID=0100DCA0064A7002
00:00:04.457 |I| HLE.OsThread.8 Loader OpenDataStorageByDataId: Opened AddOnContent Data TitleID=0100DCA0064A7002
00:00:04.501 |I| HLE.OsThread.22 SurfaceFlinger CreateLayerFromId: Creating layer 1
00:00:09.773 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:00:34.857 |I| HLE.OsThread.7 AudioRenderer AcquireSessionId: Registered new output (0)
00:00:34.902 |I| HLE.OsThread.7 Audio EnsureAudioStreamSetup: New audio stream setup with a target sample count of 1024
00:00:39.037 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:00:41.720 |I| HLE.OsThread.22 ServiceAm .ctor: Applet 'PlayerSelect' created.
00:01:03.291 |I| .NET ThreadPool Worker Ptc Save: Saved Profiling Info (size: 46390 bytes, profiled functions: 14343).
00:01:14.154 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:25.014 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:26.442 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:26.919 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:33.291 |I| .NET ThreadPool Worker Ptc Save: Saved Profiling Info (size: 60598 bytes, profiled functions: 18310).
00:01:37.894 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:42.286 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:43.194 |W| HLE.OsThread.10 ServiceNv Wait: GPU processing thread is too slow, waiting on CPU...
00:01:50.476 |I| HLE.OsThread.7 AudioRenderer ReleaseSessionId: Unregistered output (0)
00:01:50.476 |I| HLE.OsThread.7 AudioRenderer AcquireSessionId: Registered new output (0)
00:01:50.487 |I| HLE.OsThread.22 ServiceAm SetTerminateResult: Result = 0x000002a2 (ResultErrApplicationAborted (2162-0001)).
00:01:51.137 |I| HLE.GuestThread.41 Cpu PrintGuestStackTrace: Guest stack trace:
Process: Luigi's Mansion 3, PID: 98
   0x0000000009e99d3c	subsdk0:0x17ed3c	
   0x0000000009e99e08	subsdk0:0x17ee08	
   0x0000000009e99eb0	subsdk0:0x17eeb0	nn::diag::detail::AbortImpl(nn::diag::MetaLogContext const&, nn::Result const*, char const*, ...)
   0x0000000009db277c	subsdk0:0x9777c	
   0x0000000008a2cc60	main:0xa28c60	
   0x00000000089c0dc8	main:0x9bcdc8	
   0x00000000089c094c	main:0x9bc94c	
   0x00000000089d87ac	main:0x9d47ac	
   0x00000000089d8298	main:0x9d4298	
   0x000000000a044e88	subsdk0:0x329e88	
   0x000000000a049294	subsdk0:0x32e294	
   0x0000000000000000


00:01:51.138 |I| HLE.GuestThread.41 Cpu PrintGuestRegisterPrintout: Guest CPU registers:
	X[00]:	0x0000000000000000
	X[01]:	0x00000011fe171b58 (SP:-0x01c8)
	X[02]:	0x0000000000000004
	X[03]:	0x0000000000000000
	X[04]:	0x0000000000000000
	X[05]:	0x0000000000000000
	X[06]:	0x0000000000000000
	X[07]:	0x0000000000000000
	X[08]:	0x00000000002a8043
	X[09]:	0x0000000000000000
	X[10]:	0x0000000000000018
	X[11]:	0x0080000000000000
	X[12]:	0x0000000000000000
	X[13]:	0x000000000000000e
	X[14]:	0x0000000000000002
	X[15]:	0x0000000000000000
	X[16]:	0x0000000009e9b1d0 (subsdk0:0x1801d0)	=> nn::diag::detail::Abort(nn::Result const*)
	X[17]:	0x000000000a9210d0
	X[18]:	0x0000000000000000
	X[19]:	0x00000011fe171b58 (SP:-0x01c8)
	X[20]:	0x0000000000000000
	X[21]:	0x0000000000000000
	X[22]:	0x000000000a6e4f6d
	X[23]:	0x000000000a6e4f6d
	X[24]:	0x000000000a6e4f6d
	X[25]:	0x000000000a6e4f6d
	X[26]:	0x00000011fe171b20 (SP:-0x0190)
	X[27]:	0x0000000000000018
	X[28]:	0x00000000097c1000
	FP:	0x00000011fe171990 (SP)
	LR:	0x0000000009e9b1e8 (subsdk0:0x1801e8)	=> nn::diag::detail::Abort(nn::Result const*):0x0018
	SP:	0x00000011fe171990 (SP)
	PC:	0x000000000a099ba4 (subsdk0:0x37eba4)	=> 


00:01:51.150 |W| HLE.GuestThread.24 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.36 KernelSvc : WaitProcessWideKeyAtomic() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.23 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.37 KernelSvc : WaitProcessWideKeyAtomic() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.27 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.29 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.38 KernelSvc : WaitProcessWideKeyAtomic() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.39 KernelSvc : WaitProcessWideKeyAtomic() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.40 KernelSvc : WaitProcessWideKeyAtomic() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.28 KernelSvc : WaitSynchronization(handleIndex: 0xFFFFFFFF) = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.30 KernelSvc : WaitSynchronization(handleIndex: 0xFFFFFFFF) = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.31 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.32 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.33 KernelSvc : SendSyncRequest() = TerminationRequested
00:01:51.150 |W| HLE.GuestThread.34 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.151 |W| HLE.GuestThread.35 KernelSvc : WaitForAddress() = TerminationRequested
00:01:51.237 |E| HLE.GuestThread.41 Application : Unhandled exception caught: Ryujinx.HLE.Exceptions.GuestBrokeExecutionException: The guest program broke execution!
   at Ryujinx.HLE.HOS.Kernel.SupervisorCall.Syscall.Break(UInt64 reason) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.HLE\HOS\Kernel\SupervisorCall\Syscall.cs:line 1878
   at Ryujinx.HLE.HOS.Kernel.SupervisorCall.SyscallDispatch.Break64(Syscall syscall, IExecutionContext context) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.HLE\Ryujinx.Horizon.Kernel.Generators\Ryujinx.Horizon.Generators.Kernel.SyscallGenerator\SyscallDispatch.g.cs:line 1807
   at Ryujinx.HLE.HOS.Kernel.SupervisorCall.SyscallDispatch.Dispatch64(Syscall syscall, IExecutionContext context, Int32 id) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.HLE\Ryujinx.Horizon.Kernel.Generators\Ryujinx.Horizon.Generators.Kernel.SyscallGenerator\SyscallDispatch.g.cs:line 3569
   at Ryujinx.HLE.HOS.Kernel.SupervisorCall.SyscallHandler.SvcCall(IExecutionContext context, UInt64 address, Int32 id) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.HLE\HOS\Kernel\SupervisorCall\SyscallHandler.cs:line 17
   at ARMeilleure.Instructions.NativeInterface.SupervisorCall(UInt64 address, Int32 imm) in D:\a\Ryujinx\Ryujinx\src\ARMeilleure\Instructions\NativeInterface.cs:line 50
   at ARMeilleure.Translation.Translator.Execute(ExecutionContext context, UInt64 address) in D:\a\Ryujinx\Ryujinx\src\ARMeilleure\Translation\Translator.cs:line 155
   at Ryujinx.Cpu.Jit.JitCpuContext.Execute(IExecutionContext context, UInt64 address) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.Cpu\Jit\JitCpuContext.cs:line 32
   at Ryujinx.HLE.HOS.ArmProcessContext`1.Execute(IExecutionContext context, UInt64 codeAddress) in D:\a\Ryujinx\Ryujinx\src\Ryujinx.HLE\HOS\ArmProcessContext.cs:line 50
   at Ryujinx.HLE.HOS.Kernel.Threading.KThread.ThreadStart() in D:\a\Ryujinx\Ryujinx\src\Ryujinx.HLE\HOS\Kernel\Threading\KThread.cs:line 1266
```

**Comment by MetrosexualGarbodor** [comment]
Duplicate of #207
